content-test/addon/PageModProvider.test.js

```diff
@@ -0,0 +1,182 @@
+      });
+      it("should add the worker to .workers", () => {
+        setup();
+        assert.equal(instance.workers.size, 1);
+        assert.ok(instance.workers.get(worker));
+      });
+      it("should listen to detach, and remove worker when it is detached", () => {
+        setup();
+        assert.ok(instance.workers.get(worker));
+        const [eventName, callback] = worker.on.firstCall.args;
+        assert.equal(eventName, "detach", "added callback to 'detach'");
+        callback();
+        assert.isUndefined(instance.workers.get(worker), "callback removes worker");
+      });
+      it("should listen to CONTENT_TO_ADDON event from worker port", () => {
+        setup();
+        assert.calledWith(worker.port.on, CONTENT_TO_ADDON);
+        assert.isFunction(worker.port.on.firstCall.args[1]);
+      });
+      describe("worker.on(CONTENT_TO_ADDON)", () => {
+        let onMessage;
+        function setupWithOnMessage() {
+          onMessage = sinon.spy();
+          setup({options: {onMessage}});
+          // This is the callback for a CONTENT_TO_ADDON message
+          return worker.port.on.firstCall.args[1];
+        }
+        it("should remove the worker if the event type is 'pagehide'", () => {
+          const callback = setupWithOnMessage();
+          assert.ok(instance.workers.get(worker));
+          callback({type: "pagehide"});
+          assert.isUndefined(instance.workers.get(worker), "pagehide event removes worker");
+        });
+        it("should remove the worker if the event type is 'pagehide'", () => {
+          const callback = setupWithOnMessage();
+          callback({type: "pagehide"});
+          assert.isUndefined(instance.workers.get(worker), "pagehide event removes worker");
+        });
+        it("should call onMessage if defined", () => {
+          const callback = setupWithOnMessage();
+          callback({type: "FOO"});
+          assert.calledOnce(onMessage);
+        });
+        it("should call onMessage with the right message", () => {
+          const callback = setupWithOnMessage();
+          callback({type: "FOO", data: "bar"});
+          const result = onMessage.firstCall.args[0];
+          assert.deepEqual(result.msg, {
+            type: "FOO",
+            data: "bar",
+            workerId: instance.workers.get(worker)
+          });
+        });
+        it("should call onMessage with the right worker", () => {
+          const callback = setupWithOnMessage();
+          callback({type: "FOO"});
+          const result = onMessage.firstCall.args[0];
+          assert.equal(result.worker, worker);
+        });
+      });
+    });
+  });
+  describe("#getWorkerById", () => {
+    it("should get a worker by id", () => {
+      const id = "asdkl123";
+      const worker = {};
+      instance.workers.set(worker, id);
+      instance.workers.set({}, "wjklwlk");
+      instance.workers.set({}, "a3elaks");
+      assert.equal(instance.getWorkerById(id), worker);
```

Comment:
It's not super clear to me how this test is testing that it got the correct worker based on the id... if all the workers are just an empty object, they'll all pass this test no matter what id you pass in? Unless you're just testing that the ```getWorkerById``` function returns any worker?